src/test/java/org/opensearch/knn/index/query/KNNQueryFactoryTests.java

```diff
@@ -120,4 +125,34 @@ public void testCreateFaissQueryWithFilter_withValidValues_thenSuccess() {
         assertEquals(testK, ((KNNQuery) query).getK());
         assertEquals(FILTER_QUERY, ((KNNQuery) query).getFilterQuery());
     }
+
+    public void testCreate_whenLuceneWithParentFilter_thenReturnDiversifyingQuery() {
+        validateDiversifyingQueryWithParentFilter(VectorDataType.BYTE, DiversifyingChildrenByteKnnVectorQuery.class);
+        validateDiversifyingQueryWithParentFilter(VectorDataType.FLOAT, DiversifyingChildrenFloatKnnVectorQuery.class);
```

Comment:
is there a test for negation of the if statement, when parent filter is null? I checked existing tests quickly, seems that we are checking using `isAssignableFrom`, which is questionable as lucene new query class extends the old on, like `DiversifyingChildrenByteKnnVectorQuery` extends `KnnByteVectorQuery`

Reply:
Good catch. Need to switch from `assignableFrom` to equality check.